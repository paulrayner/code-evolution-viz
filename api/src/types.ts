/**
 * Processor Data Types (shared with processor/src/types.ts)
 * These are copied here to avoid cross-directory TypeScript imports
 */

export interface FileNode {
  path: string;
  name: string;
  type: 'file';
  loc: number;
  extension: string;
  lastModified: string | null;
  lastAuthor: string | null;
  lastCommitHash: string | null;
  commitCount: number | null;
  contributorCount: number | null;
  firstCommitDate: string | null;
  recentLinesChanged: number | null;
  avgLinesPerCommit: number | null;
  daysSinceLastModified: number | null;
  isGenerated?: boolean;
}

export interface DirectoryNode {
  path: string;
  name: string;
  type: 'directory';
  children: TreeNode[];
}

export type TreeNode = FileNode | DirectoryNode;

export interface RepositorySnapshot {
  repositoryPath: string;
  commit: string;
  timestamp: string;
  author: string;
  message: string;
  tree: DirectoryNode;
  commitMessages: Record<string, string>;
  stats: {
    totalFiles: number;
    totalLoc: number;
    filesByExtension: Record<string, number>;
  };
}

export interface TimelineData {
  format: 'timeline-v1';
  repositoryPath: string;
  headSnapshot: RepositorySnapshot;
  timeline: any;
}

export interface TimelineDataV2 {
  format: 'timeline-v2';
  repositoryPath: string;
  metadata: any;
  commits: any[];
}

/**
 * API Response Types for CodeCohesion API
 */

/**
 * HATEOAS Link Types
 */

export interface Link {
  href: string;
  description?: string;
  templated?: boolean;
}

export interface Links {
  [rel: string]: Link;
}

export interface Action {
  method: string;
  href: string;
  description: string;
}

export interface Actions {
  [name: string]: Action;
}

export interface RepoListItem {
  id: string;
  name: string;
  format?: string;
  _links?: Links;
  _actions?: Actions;
}

export interface RepoInfo extends RepoListItem {
  url?: string;
}

export interface ReposResponse {
  repos: RepoListItem[];
  _links?: Links;
}

export interface StatsResponse {
  repository: {
    id: string;
    path: string;
  };
  analyzedAt: string;
  commit: string;
  stats: {
    totalFiles: number;
    totalLoc: number;
    filesByExtension: Record<string, number>;
  };
}

export interface ContributorInfo {
  email: string;
  filesChanged: number;
  lastModified: string;
}

export interface ContributorsResponse {
  repository: {
    id: string;
    url?: string;
  };
  period: {
    since?: string;
    until?: string;
    days?: number;
  };
  contributors: ContributorInfo[];
  total: number;
}

export interface FilesResponse {
  files: FileNode[];
  total: number;
}

export interface HotspotsResponse {
  topChurn: FileNode[];
  topContributors: FileNode[];
}

export interface HelpAction {
  description: string;
  method: string;
  url: string;
  example?: string;
}

export interface ErrorResponse {
  error: string;
  code?: string;
  message?: string;
  details?: any;
  help?: {
    message?: string;
    actions?: HelpAction[];
  };
  docs?: string;
}

export interface Example {
  description: string;
  request: string;
}

export interface RootResponse {
  service: string;
  version: string;
  description: string;
  _links: Links;
  examples: {
    [name: string]: Example;
  };
  capabilities: string[];
}

export interface HealthResponse {
  status: string;
  uptime: number;
  timestamp: string;
}
