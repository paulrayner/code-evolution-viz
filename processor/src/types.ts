/**
 * Shared types for repository analysis
 */

export interface FileNode {
  path: string;
  name: string;
  type: 'file';
  loc: number;
  extension: string;
  lastModified: string | null;
  lastAuthor: string | null;
  lastCommitHash: string | null;
}

export interface DirectoryNode {
  path: string;
  name: string;
  type: 'directory';
  children: TreeNode[];
}

export type TreeNode = FileNode | DirectoryNode;

export interface RepositorySnapshot {
  repositoryPath: string;
  commit: string;
  timestamp: string;
  author: string;
  message: string;
  tree: DirectoryNode;
  commitMessages: Record<string, string>; // Map of commit hash -> commit message
  stats: {
    totalFiles: number;
    totalLoc: number;
    filesByExtension: Record<string, number>;
  };
}
