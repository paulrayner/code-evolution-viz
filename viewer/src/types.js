/**
 * Shared types (mirrors processor types)
 */
export {};
