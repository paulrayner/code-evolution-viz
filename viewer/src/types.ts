/**
 * Shared types (mirrors processor types)
 */

export interface FileNode {
  path: string;
  name: string;
  type: 'file';
  loc: number;
  extension: string;
  lastModified: string | null;
  lastAuthor: string | null;
  lastCommitHash: string | null;
  commitCount: number | null;        // Total commits for this file (churn)
  contributorCount: number | null;   // Number of unique contributors
  firstCommitDate: string | null;    // Date of first commit (file age)
  recentLinesChanged: number | null; // Lines added + deleted in last 90 days
  avgLinesPerCommit: number | null;  // Average lines changed per commit (volatility)
  daysSinceLastModified: number | null; // Days since last modification
}

export interface DirectoryNode {
  path: string;
  name: string;
  type: 'directory';
  children: TreeNode[];
}

export type TreeNode = FileNode | DirectoryNode;

export interface RepositorySnapshot {
  repositoryPath: string;
  commit: string;
  timestamp: string;
  author: string;
  message: string;
  tree: DirectoryNode;
  commitMessages: Record<string, string>; // Map of commit hash -> commit message
  stats: {
    totalFiles: number;
    totalLoc: number;
    filesByExtension: Record<string, number>;
  };
}

/**
 * Timeline format with adaptive sampling (Slice 2)
 */
export interface TimelineData {
  format: 'timeline-v1';
  repositoryPath: string;
  headSnapshot: RepositorySnapshot;  // Backward compatibility - same as static format
  timeline: {
    totalCommits: number;
    dateRange: {
      first: string;
      last: string;
    };
    baseSampling: {
      algorithm: 'adaptive-v2';
      targetCount: number;
      actualCount: number;
      commits: CommitSnapshot[];
    };
    drillDownLayers?: DrillDownLayer[];  // Optional - for time-range detail
  };
}

/**
 * Single commit snapshot with delta information
 */
export interface CommitSnapshot {
  hash: string;
  date: string;
  author: string;
  message: string;
  tags: string[];
  isMergeCommit: boolean;
  importanceScore: number;
  changes: {
    filesAdded: string[];
    filesModified: string[];
    filesDeleted: string[];
    totalFilesChanged: number;
    linesAdded: number;
    linesDeleted: number;
  };
  tree?: DirectoryNode;  // Optional - full tree for keyframes only
}

/**
 * Drill-down layer for detailed time range analysis
 */
export interface DrillDownLayer {
  id: string;
  label: string;
  dateRange: {
    start: string;
    end: string;
  };
  commits: CommitSnapshot[];  // All commits in this range
}
