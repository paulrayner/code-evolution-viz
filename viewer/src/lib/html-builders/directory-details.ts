/**
 * Pure HTML builder for directory details panel
 * Extracted from main.ts showDirectoryDetails function
 */

import { DirectoryNode } from '../../types';

export interface DirectoryDetailsData {
  dir: DirectoryNode;
  stats: {
    totalLoc: number;
    fileCount: number;
    dirCount: number;
    dominantExt: string;
    dominantName: string;
  };
  lastModified: {
    date: string; // Formatted date string
    author: string;
  };
  githubDirUrl: string | null;
}

/**
 * Build HTML for directory details panel
 * Pure function - no side effects, no DOM access, no global state
 */
export function buildDirectoryDetailsHTML(data: DirectoryDetailsData): string {
  const { dir, stats, lastModified, githubDirUrl } = data;

  return `
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">${dir.path || '(root)'}</span>
    </div>
    ${githubDirUrl ? `<div class="info-row">
      <span class="label">View on GitHub</span>
      <span class="value"><a href="${githubDirUrl}" target="_blank" style="color: #4a9eff; text-decoration: none;">ðŸ”— Open folder</a></span>
    </div>` : ''}
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">${stats.totalLoc.toLocaleString()}</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">${stats.fileCount}</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">${stats.dirCount}</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">${stats.dominantName}</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">${lastModified.date}</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">${lastModified.author}</span>
    </div>
  `;
}
