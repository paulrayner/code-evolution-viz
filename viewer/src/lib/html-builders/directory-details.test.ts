import { describe, it, expect } from 'vitest';
import { buildDirectoryDetailsHTML, DirectoryDetailsData } from './directory-details';
import { DirectoryNode } from '../../types';

// Helper function to create mock directory nodes
function createMockDir(overrides: Partial<DirectoryNode> = {}): DirectoryNode {
  return {
    path: overrides.path || '/src',
    name: overrides.name || 'src',
    type: 'directory',
    children: overrides.children || [],
  };
}

describe('buildDirectoryDetailsHTML', () => {
  it('should render complete directory details with all data present', () => {
    const dir = createMockDir({ path: '/src/components', name: 'components' });
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 5420,
        fileCount: 15,
        dirCount: 3,
        dominantExt: 'tsx',
        dominantName: 'TypeScript React',
      },
      lastModified: {
        date: 'Oct 20, 2025',
        author: 'Jane Smith',
      },
      githubDirUrl: 'https://github.com/test/repo/tree/main/src/components',
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
  });

  it('should render directory details without GitHub URL', () => {
    const dir = createMockDir({ path: '/lib', name: 'lib' });
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 2100,
        fileCount: 8,
        dirCount: 2,
        dominantExt: 'ts',
        dominantName: 'TypeScript',
      },
      lastModified: {
        date: 'Sep 15, 2025',
        author: 'Bob Johnson',
      },
      githubDirUrl: null,
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
    expect(html).not.toContain('View on GitHub');
  });

  it('should render root directory details', () => {
    const dir: DirectoryNode = {
      path: '',
      name: 'root',
      type: 'directory',
      children: [],
    };
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 15000,
        fileCount: 50,
        dirCount: 10,
        dominantExt: 'ts',
        dominantName: 'TypeScript',
      },
      lastModified: {
        date: 'Oct 22, 2025',
        author: 'Alice Dev',
      },
      githubDirUrl: 'https://github.com/test/repo',
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
    expect(html).toContain('(root)');
  });

  it('should render empty directory details', () => {
    const dir = createMockDir({ path: '/empty', name: 'empty' });
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 0,
        fileCount: 0,
        dirCount: 0,
        dominantExt: 'none',
        dominantName: 'none',
      },
      lastModified: {
        date: 'Unknown',
        author: 'Unknown',
      },
      githubDirUrl: null,
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
    expect(html).toContain('0');
    expect(html).toContain('Unknown');
  });

  it('should render directory with large LOC count formatted correctly', () => {
    const dir = createMockDir({ path: '/src/large', name: 'large' });
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 1234567,
        fileCount: 250,
        dirCount: 15,
        dominantExt: 'js',
        dominantName: 'JavaScript',
      },
      lastModified: {
        date: 'Oct 1, 2025',
        author: 'Dev Team',
      },
      githubDirUrl: null,
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
    expect(html).toContain('1,234,567');
  });

  it('should render directory with single file and no subdirectories', () => {
    const dir = createMockDir({ path: '/config', name: 'config' });
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 50,
        fileCount: 1,
        dirCount: 0,
        dominantExt: 'json',
        dominantName: 'JSON',
      },
      lastModified: {
        date: 'Aug 10, 2025',
        author: 'Config Bot',
      },
      githubDirUrl: 'https://github.com/test/repo/tree/main/config',
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
  });

  it('should render directory with only subdirectories', () => {
    const dir = createMockDir({ path: '/modules', name: 'modules' });
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 8000,
        fileCount: 0,
        dirCount: 5,
        dominantExt: 'ts',
        dominantName: 'TypeScript',
      },
      lastModified: {
        date: 'Oct 18, 2025',
        author: 'Module Team',
      },
      githubDirUrl: null,
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
    expect(html).toContain('0</span>'); // 0 files
  });

  it('should render directory with unknown dominant type', () => {
    const dir = createMockDir({ path: '/misc', name: 'misc' });
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 500,
        fileCount: 10,
        dirCount: 1,
        dominantExt: 'xyz',
        dominantName: 'xyz',
      },
      lastModified: {
        date: 'Jul 5, 2025',
        author: 'Unknown',
      },
      githubDirUrl: null,
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
    expect(html).toContain('xyz');
  });

  it('should handle special characters in path and name', () => {
    const dir = createMockDir({ path: '/special-chars/foo_bar', name: 'foo_bar' });
    const data: DirectoryDetailsData = {
      dir,
      stats: {
        totalLoc: 300,
        fileCount: 5,
        dirCount: 1,
        dominantExt: 'py',
        dominantName: 'Python',
      },
      lastModified: {
        date: 'Oct 10, 2025',
        author: "O'Brien",
      },
      githubDirUrl: 'https://github.com/test/repo/tree/main/special-chars/foo_bar',
    };

    const html = buildDirectoryDetailsHTML(data);
    expect(html).toMatchSnapshot();
    expect(html).toContain('foo_bar');
    expect(html).toContain("O'Brien");
  });
});
