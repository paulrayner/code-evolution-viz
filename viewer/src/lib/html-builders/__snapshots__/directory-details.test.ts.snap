// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildDirectoryDetailsHTML > should handle special characters in path and name 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">/special-chars/foo_bar</span>
    </div>
    <div class="info-row">
      <span class="label">View on GitHub</span>
      <span class="value"><a href="https://github.com/test/repo/tree/main/special-chars/foo_bar" target="_blank" style="color: #4a9eff; text-decoration: none;">ðŸ”— Open folder</a></span>
    </div>
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">300</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">5</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">1</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">Python</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Oct 10, 2025</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">O'Brien</span>
    </div>
  "
`;

exports[`buildDirectoryDetailsHTML > should render complete directory details with all data present 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">/src/components</span>
    </div>
    <div class="info-row">
      <span class="label">View on GitHub</span>
      <span class="value"><a href="https://github.com/test/repo/tree/main/src/components" target="_blank" style="color: #4a9eff; text-decoration: none;">ðŸ”— Open folder</a></span>
    </div>
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">5,420</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">15</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">3</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">TypeScript React</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Oct 20, 2025</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">Jane Smith</span>
    </div>
  "
`;

exports[`buildDirectoryDetailsHTML > should render directory details without GitHub URL 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">/lib</span>
    </div>
    
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">2,100</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">8</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">2</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">TypeScript</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Sep 15, 2025</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">Bob Johnson</span>
    </div>
  "
`;

exports[`buildDirectoryDetailsHTML > should render directory with large LOC count formatted correctly 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">/src/large</span>
    </div>
    
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">1,234,567</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">250</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">15</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">JavaScript</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Oct 1, 2025</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">Dev Team</span>
    </div>
  "
`;

exports[`buildDirectoryDetailsHTML > should render directory with only subdirectories 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">/modules</span>
    </div>
    
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">8,000</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">0</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">5</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">TypeScript</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Oct 18, 2025</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">Module Team</span>
    </div>
  "
`;

exports[`buildDirectoryDetailsHTML > should render directory with single file and no subdirectories 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">/config</span>
    </div>
    <div class="info-row">
      <span class="label">View on GitHub</span>
      <span class="value"><a href="https://github.com/test/repo/tree/main/config" target="_blank" style="color: #4a9eff; text-decoration: none;">ðŸ”— Open folder</a></span>
    </div>
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">50</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">1</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">0</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">JSON</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Aug 10, 2025</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">Config Bot</span>
    </div>
  "
`;

exports[`buildDirectoryDetailsHTML > should render directory with unknown dominant type 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">/misc</span>
    </div>
    
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">500</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">10</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">1</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">xyz</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Jul 5, 2025</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">Unknown</span>
    </div>
  "
`;

exports[`buildDirectoryDetailsHTML > should render empty directory details 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">/empty</span>
    </div>
    
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">0</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">0</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">0</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">none</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Unknown</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">Unknown</span>
    </div>
  "
`;

exports[`buildDirectoryDetailsHTML > should render root directory details 1`] = `
"
    <div class="info-row">
      <span class="label">Type</span>
      <span class="value">Directory</span>
    </div>
    <div class="info-row">
      <span class="label">Path</span>
      <span class="value">(root)</span>
    </div>
    <div class="info-row">
      <span class="label">View on GitHub</span>
      <span class="value"><a href="https://github.com/test/repo" target="_blank" style="color: #4a9eff; text-decoration: none;">ðŸ”— Open folder</a></span>
    </div>
    <div class="info-row">
      <span class="label">Total LOC</span>
      <span class="value">15,000</span>
    </div>
    <div class="info-row">
      <span class="label">Files</span>
      <span class="value">50</span>
    </div>
    <div class="info-row">
      <span class="label">Subdirectories</span>
      <span class="value">10</span>
    </div>
    <div class="info-row">
      <span class="label">Dominant Type</span>
      <span class="value">TypeScript</span>
    </div>
    <div class="info-row">
      <span class="label">Last Modified</span>
      <span class="value">Oct 22, 2025</span>
    </div>
    <div class="info-row">
      <span class="label">Last Author</span>
      <span class="value">Alice Dev</span>
    </div>
  "
`;
